File: app/src/main/java/com/shindra/TramLines/Trams_View_Holder.java
package com.shindra.TramLines;

import android.widget.Button;
import android.widget.ImageView;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.shindra.R;
import com.shindra.ctslibrary.bo.Line;

public class Trams_View_Holder extends RecyclerView.ViewHolder
{
    //private final ImageView tram_view;
    private final ImageView tram_logo;
    //private final Button button;

    public Trams_View_Holder(@NonNull View itemView)
    {
        super(itemView);
        this.tram_logo = itemView.findViewById(R.id.Tram_Logo);
        //this.tram_view = itemView.findViewById(R.id.Tram_Line_View);
        //this.button = itemView.findViewById(R.id.Tram_view_button);
    }

    public void onBind_logo(Line image)
    {
        switch (image.getName())
        {
            case "A":
                tram_logo.setImageResource(R.drawable.tram_a);
                break;

            case "B":
                tram_logo.setImageResource(R.drawable.tram_b);
                break;

            case "C":
                tram_logo.setImageResource(R.drawable.tram_c);
                break;

            case "D":
                tram_logo.setImageResource(R.drawable.tram_d);
                break;

            case "E":
                tram_logo.setImageResource(R.drawable.tram_e);
                break;

            case "F":
                tram_logo.setImageResource(R.drawable.tram_f);
                break;

            default:
                tram_logo.setImageResource(R.drawable.tram);
                break;
        }
    }
}

File: app/src/main/java/com/shindra/Lignes/TramsViewHolder.kt
package com.shindra.Lignes

import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.shindra.R

class TramsViewHolder(GenericCardView: View) : RecyclerView.ViewHolder(GenericCardView){
    val TramLineImage: ImageView = GenericCardView.findViewById(R.id.TramLineImageView);
    val ButtonHoraires: Button = GenericCardView.findViewById(R.id.BoutonHoraires);

    fun PutTramInfoIntoCardView(TramLineLetter : String) {
        TramLineImage.setImageResource(when(TramLineLetter){
            "A" -> R.drawable.ic_tram_a
            "B" -> R.drawable.ic_tram_b
            "C" -> R.drawable.ic_tram_c
            "D" -> R.drawable.ic_tram_d
            "E" -> R.drawable.ic_tram_e
            "F" -> R.drawable.ic_tram_f
            else -> {
                R.drawable.ic_tram;
            }
        })
        }
    }